src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-cluster-form/nfs-cluster-form.component.ts

```diff
@@ -0,0 +1,110 @@
+import { NotificationType } from '~/app/shared/enum/notification-type.enum';
+import { NotificationService } from '~/app/shared/services/notification.service';
+import { NFSBwIopConfig } from '../models/nfs-cluster-config';
+import { CdForm } from '~/app/shared/forms/cd-form';
+
+@Component({
+  selector: 'cd-nfs-cluster-form',
+  templateUrl: './nfs-cluster-form.component.html',
+  styleUrls: ['./nfs-cluster-form.component.scss']
+})
+export class NfsClusterFormComponent extends CdForm implements OnInit {
+  @ViewChild(NfsRateLimitComponent, { static: false })
+  nfsRateLimitComponent!: NfsRateLimitComponent;
+
+  action: string;
+  nfsForm: CdFormGroup;
+  resource: string;
+  isEdit = false;
+  disablePerShare = true;
+  disablePerClient = true;
+  disablePerSharePerClient = true;
+  fsal: SUPPORTED_FSAL;
+  id: string;
+  qosValue: string;
+  nfsBwIopConfig: NFSBwIopConfig;
+  submitObservables: Observable<Object>[] = [];
+
+  constructor(
+    public actionLabels: ActionLabelsI18n,
+    private route: ActivatedRoute,
+    private router: Router,
+    private nfsService: NfsService,
+    private notificationService: NotificationService
+  ) {
+    super();
+    this.resource = $localize`Cluster`;
+  }
+  ngOnInit(): void {
+    this.createClusterForm();
+    this.id = this.route.snapshot.paramMap.get('cluster_id');
+    this.fsal = getFsalFromRoute(this.router.url);
+    if (this.router.url.startsWith(`/${getPathfromFsal(this.fsal)}/nfs/edit`)) {
+      this.isEdit = true;
+    }
+    if (this.isEdit) {
+      this.action = this.actionLabels.EDIT;
+      this.nfsService.getClusterBandwidthOpsConfig(this.id).subscribe((data: NFSBwIopConfig) => {
+        this.nfsBwIopConfig = data;
+        this.nfsForm.get('cluster_id').setValue(this.id);
+      });
+      this.loadingReady();
+    } else {
+      this.action = this.actionLabels.CREATE;
+    }
+  }
+  childCompErrorHandler(event: Event) {
+    this.nfsForm.addControl('rateLimit', event);
+  }
+  createClusterForm() {
+    this.nfsForm = new CdFormGroup({
+      cluster_id: new UntypedFormControl({ value: '', disabled: true })
+    });
+  }
+  goToListView() {
+    this.router.navigate([`/${getPathfromFsal(this.fsal)}/nfs`]);
+  }
+  submitAction() {
+    let notificationTitle: string;
+    // Exit immediately if the form isn't dirty.
+    if (this.nfsForm.pristine && this.nfsRateLimitComponent.rateLimitForm.pristine) {
+      this.goToListView();
+      return;
+    }
+    let clusteFormrObj = this.nfsForm.getRawValue();
+    delete clusteFormrObj.rateLimit;
+    let ratelimitValue: NFSBwIopConfig = this.nfsRateLimitComponent.getRateLimitFormValue();
+    ratelimitValue = {
+      ...ratelimitValue,
+      ...clusteFormrObj,
+      disable_qos: !ratelimitValue.enable_qos
+    };
+    delete ratelimitValue.enable_qos;
+    this.submitObservables.push(this.nfsService.enableQosBandwidthForCLuster(ratelimitValue));
+    notificationTitle = $localize`Cluster QOS Type Configured for '${this.id}'`;
```

Comment:
Its QoS btw and not QOS. Please check if thats repeated anywhere else and see if you can change it its being shown to a user.
```suggestion
    notificationTitle = $localize`Cluster QoS Type Configured for '${this.id}'`;